.evergreen/serverless/create-instance.sh

```diff
@@ -46,94 +61,59 @@ curl \
 
     if [ "$STATE_NAME" = "IDLE" ]; then
         duration="$SECONDS"
-        echo "setup done! ($(($duration / 60))m $(($duration % 60))s elapsed)"
-
-        API_RESPONSE=$API_RESPONSE \
-        INSTANCE_NAME=$INSTANCE_NAME \
-        LOADBALANCED=$LOADBALANCED \
-        $PYTHON_BINARY - << EOF | tee serverless-expansion.yml
-import json
-import sys
-import os
-
-def upsert_option(uri, name, value):
-    # Add the URI option <name>=<value> to the URI if it is not already present.
-    if "?" not in uri:
-        if uri.endswith("/"):
-            return uri + "?" + name + "=" + value
-        else:
-            return uri + "/?" + name + "=" + value
-
-    option_string = uri[uri.find("?")+1:]
-    options = option_string.split("&")
-    option_names = [option.split("=")[0] for option in options]
-    if name in option_names:
-        return uri
-    else:
-        return uri + "&" + name + "=" + value
-
-def select_last_host(uri):
-    return "mongodb://" + uri[uri.rfind(",")+1:]
-
-api_response = json.loads(os.environ["API_RESPONSE"])
-
-# The srvAddress response field is an SRV URI pointing to the load balancer. A
-# corresponding TXT record includes the loadBalanced=true URI option. This will
-# be used for MULTI_ATLASPROXY_SERVERLESS_URI.
-mongodb_srv_uri = api_response['srvAddress']
-multi_atlasproxy_serverless_uri = mongodb_srv_uri
-
-# The mongoURI response field reports the serverless instance(s) behind the load
-# balancer. This script will reduce it to a single host and append necessary URI
-# options to construct SINGLE_ATLASPROXY_SERVERLESS_URI, which is necessary for
-# testing fail points (much like SINGLE_LB_MONGOS_URI).
-mongodb_uri = api_response['mongoURI']
-single_atlasproxy_serverless_uri = select_last_host(mongodb_uri)
-single_atlasproxy_serverless_uri = upsert_option(single_atlasproxy_serverless_uri, "loadBalanced", "true")
-single_atlasproxy_serverless_uri = upsert_option(single_atlasproxy_serverless_uri, "tls", "true")
-
-if os.environ.get("LOADBALANCED") == "ON":
-    mongodb_uri = upsert_option(mongodb_uri, "loadBalanced", "true")
-
-print('MONGODB_URI: "%s"' % (mongodb_uri))
-print('MONGODB_SRV_URI: "%s"' % (mongodb_srv_uri))
-print('SERVERLESS_INSTANCE_NAME: "%s"' % (os.environ["INSTANCE_NAME"]))
-print('SSL: ssl')
-print('AUTH: auth')
-print('TOPOLOGY: sharded_cluster')
-print('SERVERLESS: serverless')
-print('MULTI_ATLASPROXY_SERVERLESS_URI: "%s"' % (multi_atlasproxy_serverless_uri))
-print('SINGLE_ATLASPROXY_SERVERLESS_URI: "%s"' % (single_atlasproxy_serverless_uri))
+        echo "Setup done! ($(($duration / 60))m $(($duration % 60))s elapsed)"
+
+        SERVERLESS_URI=`echo $API_RESPONSE | $PYTHON_BINARY -c "import sys, json; print(json.load(sys.stdin)['connectionStrings']['standardSrv'])" | tr -d '\r\n'`
+
+        SERVERLESS_URI=$SERVERLESS_URI \
+        SERVERLESS_INSTANCE_NAME=$SERVERLESS_INSTANCE_NAME \
+        echo << EOF | tee serverless-expansion.yml
```

Comment:
```suggestion
        cat << EOF | tee serverless-expansion.yml
```

Running `create-instance.sh` in zsh locally outputs an empty serverless-expansion.yml.

Reply:
Can do. If we're using `cat`, though, I expect there's no reason to use `tee`.

Also curious how you ran this in `zsh` if the script starts with `#!/bin/bash`. Were you explicitly running it through `zsh` to overcome that?

Reply:
This should now be fixed. Feel free to test/resolve.